### PR TITLE
feat(trade): display loading state for trade confirmation

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Context: https://cowservices.slack.com/archives/C0361CDG8GP/p1707990442412109

1. Display loader in the confirmation screen instead of pending modal
2. For permittable token display pending permit signing screen and after signing display confirmation screen in loading state
3. Should work the same for swap/limit/twap

https://github.com/cowprotocol/cowswap/assets/7122625/71ffbed2-2a35-4f9f-9302-4247607af9a0

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/trade/containers/TradeConfirmModal/index.tsx [hunk 1]
```diff
@@ -123,10 +121,6 @@ function InnerComponent(props: InnerComponentProps) {
     )
   }
 
-  if (pendingTrade) {
-    return <TradeConfirmPendingContent pendingTrade={pendingTrade} onDismiss={onDismiss} />
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/trade/containers/TradeConfirmModal/index.tsx:127]: This is the only place where this component is used.

Kill `TradeConfirmPendingContent`?

**Review by alfetopito — APPROVED** [review]
Tested SWAP and LIMIT, amazing!

File: apps/cowswap-frontend/src/modules/trade/containers/TradeConfirmModal/index.tsx (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/trade/containers/TradeConfirmModal/index.tsx:127]: Deleted it, thanks!

**Comment by elena-zh** [comment]
Hey @shoom3301 , great job!
Some quesitons/issues: 
1. In the Expert mode, while signing it opens the order confirm modal. Is this ok? 
2. Permittable flow: when an approval is signed, but the order has not been places, each time when I place an order I see a blinking sign approval modal before the order confirm modal. IMO, it would be nice to remove it (both on Swap and Limit forms)
3. Safe: Swap+Approve button on the Confirm modal does not go into pending state when I send a request to sign the order
4. When a request is sent to the wallet, I see that quotes are refreshed in the confirm modal, numbers are changed. Is this ok? I think, values should not be changed and correspond to the ones that are in the wallet
image

**Comment by elena-zh** [comment]
Hey @shoom3301 , the build failed. Could you please fix this?

**Comment by shoom3301** [comment]
@elena-zh oops, sorry, fixed

**Review by anxolin — APPROVED** [review]
works great!!

**Comment by shoom3301** [comment]
@elena-zh thank you!
I've removed the loader from the top of the screen

**Pull request merged**